File: backend/src/main/java/com/team01/project/domain/notification/repository/SubscriptionRepository.java
package com.team01.project.domain.notification.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.team01.project.domain.notification.entity.Subscription;

public interface SubscriptionRepository extends JpaRepository<Subscription, Long> {
	Optional<Subscription> findByUserId(String userId);

	Optional<Subscription> findByEndpoint(String endpoint);
}

File: backend/src/main/java/com/team01/project/domain/notification/repository/SubscriptionRepository.kt
package com.team01.project.domain.notification.repository

import com.team01.project.domain.notification.entity.Subscription
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface SubscriptionRepository : JpaRepository<Subscription, Long> {
    fun findByUserId(userId: String): Optional<Subscription>

    fun findByEndpoint(endpoint: String): Optional<Subscription>
}
